src/PostgREST/App.hs

```diff
@@ -102,6 +102,19 @@ app conf reqBody req =
             (cs $ userPass u) (cs $ userRole u)
           return $ responseLBS status201
             [ jsonH
+            , (hLocation,"/postgrest/users?id=eq." <> cs (userId u))
+            ] ""
+
+    (["postgrest", "users"], "PUT") -> do
+      let user = decode reqBody :: Maybe AuthUser
```

Comment:
Technically the PUT should be to a url like `/postgrest/users?id.eq=...` rather than the underspecified `/postgrest/users`.

Check out the the logic for regular PUT where we look for the specified keys
https://github.com/begriffs/postgrest/blob/master/src/PostgREST/App.hs#L185
We'll want to do the same in this user PUT, except we want `specifiedKeys` to be exactly `['id']`. If that check fails then the result should be an http 405.

If the id specified in the url and all fields are supplied in the body (i.e. `AuthUser` correctly parses) then this function should update the user record (like it already does) or create it if it does not yet exist (which it doesn't yet do).